### PR TITLE
[Refactor] Remove post optimization loss

### PR DESCRIPTION
**Pull request by mlahariya**

### CODE REVIEW

**Review by chMoussa — APPROVED** [review]

File: qadence/ml_tools/trainer.py [hunk 1]
```diff
@@ -534,11 +534,10 @@ def run_train_batch(
         # Because the loss/metrics are returned before the optimization. To sync
         # model state and current loss/metrics we calculate them again after optimization.
         # This is not strictly necessary.
-        # TODO: Should be removed if loss can be logged at an unoptimized model state
-        with torch.no_grad():
-            post_update_loss_metrics = self.loss_fn(self.model, batch)
+        # with torch.no_grad():
```
Comment (RolandMacDoland) [inline qadence/ml_tools/trainer.py:537]: Why not simply removing it ?

**Review by RolandMacDoland — COMMENTED** [review]
Thanks @mlahariya. Why not simply removing it ?

File: qadence/ml_tools/trainer.py (hunk 1 above)
Comment (mlahariya) [reply qadence/ml_tools/trainer.py:537]: Could do that as well. Just removed it.

**Review by RolandMacDoland — APPROVED** [review]
